### ISSUE TITLE
Invert escape sequence incorrect

### ISSUE THREAD

**Issue by chrisduerr** [jwilm/alacritty#1455]
The command `echo -e "Normal \e[101;91mTest \e[7mInverted"` will print `Inverted` with the default foreground color as background and the default background color as foreground.
The correct behavior would be to print it the same as `Test`.

This bug is due to the following code:
https://github.com/jwilm/alacritty/blob/master/src/term/mod.rs#L385-L389

**Comment by chrisduerr** [comment]
Okay, I've compared `urxvt`, `XTerm` and `Alacritty`, I've excluded `Termite` because it does not invert the cell color for selection. I'd say `urxvt` has some edge-cases where things are weird and it does not handle the `Hidden` attribute at all (neither does `Alacritty` at the moment). In contrast `XTerm` is very consistent and supports the `Hidden` attribute.

In `XTerm` no matter which circumstances are present, the foreground color and background color are always inversed upon selection, even if the cell is `Hidden`, the foreground color which would be present is still used for the inverse. Rendering text with the same foreground and background color on top of each other also doesn't exhibit any special behavior, it is rendered as "invisible" (see Screenshot in previous comment) and when inversed with selection, the cells simply do not change at all (it inverses from red on red to red on red).

I think this makes the most sense, it should be easy to implement and come without any surprises for the user. I'll send a PR which implements this behavior if anyone is interested in giving it a shot.
